pkg/services/baremetal/host/host.go

```diff
@@ -1200,6 +1238,10 @@ func (s *Service) actionEnsureProvisioned() actionResult {
 
 		isTimeout, isSSHConnectionRefusedError, err := analyzeSSHOutputProvisioned(out)
 		if err != nil {
+			if errors.Is(err, errUnexpectedHostName) {
```

Comment:
Here is the code:

```
	out := sshClient.GetHostName()
	if trimLineBreak(out.StdOut) != wantHostName {
		// give the reboot some time until it takes effect
		if s.hasJustRebooted() {
			return actionContinue{delay: 2 * time.Second}
		}

		isTimeout, isSSHConnectionRefusedError, err := analyzeSSHOutputProvisioned(out)
		if err != nil {
			if errors.Is(err, errUnexpectedHostName) {
				// One possible reason: The machine gets used by a second wl-cluster
				record.Warn(s.scope.HetznerBareMetalHost, "UnexpectedHostName", err.Error())
			}
			return actionError{err: fmt.Errorf("failed to handle incomplete boot - provisioning: %w", err)}
		}
```

`trimLineBreak(out.StdOut)` could be empty because of ssh-timeout.

That's why I check for `errUnexpectedHostName`.

I want to create an Warning-Event if the host changed. This can happen if the hbmh gets used accidentally in a second cluster. I wasted some times because of this some days ago.